### ISSUE TITLE
Arduino helper functions

### ISSUE THREAD

**Issue by GoogleCodeExporter** [nanopb/nanopb#75]
```
Helper functions to obtain an istream from an Arduino Stream and an ostream 
from an Arduino Print.

I don't know if you are interested with Arduino support but I wanted to share 
it ;)
Files are located at a subfolder called "arduino".
Tested with Arduino version 1.0.4.

Sincerely,
Alejandro.

```

Original issue reported on code.google.com by `Alejandr...@gmail.com` on 23 Apr 2013 at 9:38

Attachments:
- [arduino.zip](https://storage.googleapis.com/google-code-attachments/nanopb/issue-75/comment-0/arduino.zip)

**Comment by q2dg** [comment]
Ei, what's going on with this?

**Comment by PetteriAimonen** [comment]
@q2dg Not sure what you are asking, but this issue is a few Arduino-specific helper functions that were contributed a few years ago. I don't feel like just dumping them into the nanopb code would really be that useful, but I've kept the issue open in case I or someone else ever feels like working it into a more coherent example project.

**Comment by q2dg** [comment]
Hello. Sorry for my manners. I only wanted to ask if there were some plans to create a Arduino library from this code (which was ready to import in Arduino sketches by newbies). It seems it is a low-priority need, so it's ok. Thanks

**Comment by eric-wieser** [comment]
I ended up writing my own version of this for the [nanopb-arduino](http://platformio.org/lib/show/1385/nanopb-arduino) module on PlatformIO. Looks like the code here might be more rigorous though.

**Comment by PetteriAimonen** [comment]
@eric-wieser Nice, I've added a link in README.

I think those solve the same purpose as the code initially in this issue report, so I'm closing this off.

**Comment by xu-zhiyong** [comment]
-------- 原始邮件 --------主题：Re: [nanopb/nanopb] Arduino helper functions (#75)发件人：Petteri Aimonen 收件人：nanopb/nanopb 抄送：Subscribed Closed #75.

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/nanopb/nanopb","title":"nanopb/nanopb","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/nanopb/nanopb"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"Closed #75."}],"action":{"name":"View Issue","url":"https://github.com/nanopb/nanopb/issues/75#event-1958378005"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/nanopb/nanopb/issues/75#event-1958378005",
"url": "https://github.com/nanopb/nanopb/issues/75#event-1958378005",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [nanopb/nanopb] Arduino helper functions (#75)",
"sections": [
{
"text": "",
"activityTitle": "**Petteri Aimonen**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@PetteriAimonen",
"facts": [

]
}
],
"potentialAction": [
{
"targets": [
{
"os": "default",
"uri": "https://github.com/nanopb/nanopb/issues/75#event-1958378005"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 156781298\n}"
}
],
"themeColor": "26292E"
}
]
